adafruit_lsm6ds/__init__.py

```diff
@@ -354,6 +510,37 @@ def pedometer_enable(self, enable):
         self._func_enable = enable
         self._pedometer_reset = enable
 
+    @property
+    def is_single_tap_detection_enable(self):
+        """ Whether the single tap function on the accelerometer is enabled"""
+        return (self._tap_x_enable or self._tap_y_enable or self._tap_z_enable) and (self._single_tap_on_int1_enable or self._single_tap_on_int2_enable)# pylint: disable=line-too-long
+
+    def single_tap_detection_enable(self, enable, axis_enable='xyz', int_pin=TapIntPin.INT1_PIN, threshold=TapThreshold.TAP_THRESHOLD_MID_LOW, data_rate=Rate.RATE_1_66K_HZ, accel_range=0, shock_time=TapShockTime.TAP_SHOCK_TIME_MID_HIGH, quiet_time=TapQuietTime.TAP_QUIET_TIME_MID_LOW):
+        """Enable or disable the single tap detection on the accelerometer"""
+        #Set_X_ODR(416.0f)
+        self.accelerometer_data_rate = data_rate
+        #Set_X_FS(2.0f)
+        self.accelerometer_range = accel_range
+        #LSM6DS3_ACC_GYRO_W_TAP_X_EN
+        self.tap_x_enable = 'x' in axis_enable
+        #LSM6DS3_ACC_GYRO_W_TAP_Y_EN
+        self.tap_y_enable = 'y' in axis_enable
+        #LSM6DS3_ACC_GYRO_W_TAP_Z_EN
+        self.tap_z_enable = 'z' in axis_enable
+        #Set_Tap_Threshold
+        self.tap_threshold = threshold
+        #Set_Tap_Shock_Time
+        self.tap_shock_time = shock_time
+        #Set_Tap_Quiet_Time
+        self.tap_quiet_time = quiet_time
+        if not TapIntPin.is_valid(int_pin):
```

Comment:
Because CircuitPython doesn't support concurrency at the moment, rather than enabling tap interrupts on the hardware pins,  we can read the tap interrupt status from the `TAP_SRC` and/or `ALL_INT_SRC ` registers and provide the status via a `tapped` or `events` property (see below).

We should also turn on latching with auto clear on read using the `LIR` and `INT_CLR_ON_READ` bits in the `TAP_CFG0` register so that we know if a tap event happened, not just if one is happening _right_now_ ;)

This is how we did something similar for the ADXL34x accelerometers:
https://github.com/adafruit/Adafruit_CircuitPython_ADXL34x/blob/master/adafruit_adxl34x.py#L186
in that case we had an `events` property because tap status was in the same auto-clear register as the other event types, so if we didn't report them together we would lose them.

As an alternative, we could do something similar to what the LIS3DH library is doing by providing a `tapped` property:
https://github.com/adafruit/Adafruit_CircuitPython_LIS3DH/blob/master/adafruit_lis3dh.py#L228